server/main.go

```diff
@@ -63,10 +75,21 @@ func runHandler(w http.ResponseWriter, r *http.Request) {
 
 	err = json.NewEncoder(w).Encode(output)
 	if err != nil {
-		log.Printf("failed to encode run output: %v\n", err)
+		throwE400(w, ("failed to encode run output: " + err.Error()))
+		return
 	}
 }
 
+func throwE400(w http.ResponseWriter, err string) {
```

Comment:
not sure if this is good convention for error handling, let me know if this is what you meant!